src/include/zlog/log.h

```diff
@@ -47,6 +47,12 @@ class Log {
   virtual int Trim(uint64_t position) = 0;
   virtual int trimAsync(uint64_t position, std::function<void(int)> cb) = 0;
 
+  virtual int listHeads(std::vector<std::string> &heads) = 0;
+  virtual int listHeadsAsync(std::vector<std::string> &heads, std::function<void(int)> cb) = 0;
+
+  virtual int listLinks(std::vector<std::string> &links) = 0;
+  virtual int listLinksAsync(std::vector<std::string> &links, std::function<void(int)> cb) = 0;
```

Comment:
The `zlog/log.h` header is a public header that clients use. I don't think we need to expose this publicly (yet). I think you can just remove these and leave the definitions in `LogImpl` (removing the overrides there).